src/Microsoft.AspNetCore.Authentication/PersistentAuthenticationHandler.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Antiforgery;
+using Microsoft.AspNetCore.Builder;
+
+namespace Microsoft.AspNetCore.Authentication
+{
+    public abstract class PersistentAuthenticationHandler<TOptions> : AuthenticationHandler<TOptions>
+        where TOptions : AuthenticationOptions
+    {
+        public PersistentAuthenticationHandler(IAntiforgery antiforgery)
```

Comment:
Personal request of @blowdart was that we add a base-class that enforces antiforgery. I'm not 100% sold - we could really just put all of this stuff in the base-class.

I originally started with this class overriding `HandleAuthenticateAsync`, but the problem is that you can circumvent this easily by just not calling the `base.HandleAuthenticateAsync`.

Thoughts?